src/cursor/cursor.ts

```diff
@@ -39,9 +43,58 @@ export type CursorFlag = typeof FLAGS[number];
 /** @public */
 export const FIELDS = ['numberOfRetries', 'tailableRetryInterval'] as const;
 
+/** @public */
+export interface DocumentTransforms {
+  /** Transform each document returned */
+  doc(doc: Document): Document;
+  /** Transform the value returned from the initial query */
+  query?(doc: Document): Document | Document[];
+}
+
+/** @internal */
+export interface CoreCursorPrivate {
+  /** Transforms functions */
+  transforms?: DocumentTransforms;
+  numberOfRetries: number;
+  tailableRetryInterval: number;
+  currentNumberOfRetries: number;
+  explicitlyIgnoreSession: boolean;
+  batchSize: number;
+
+  state: CursorState;
+  readConcern?: ReadConcern;
+}
+
 /** @internal */
 export type CursorPrivate = CoreCursorPrivate;
 
+/** @public Possible states for a cursor */
+export enum CursorState {
+  INIT = 0,
+  OPEN = 1,
+  CLOSED = 2,
+  GET_MORE = 3
+}
+
+/** @public */
+export interface CoreCursorOptions extends CommandOperationOptions {
```

Comment:
merge with `CursorOptions`

Reply:
Same issue as above, is it OK to delete the export from `index.ts` for this?

Reply:
yep!